tests/kernel/mem_protect/syscalls/testcase.yaml

```diff
@@ -2,3 +2,4 @@ tests:
   kernel.memory_protection.syscalls:
     filter: CONFIG_ARCH_HAS_USERSPACE
     tags: kernel security userspace ignore_faults
+    platform_exclude: qemu_cortex_r5 mps2_an521_ns
```

Comment:
Could you tell me why we skip these two platforms?

Reply:
These two boards has bug issue with processing null parameter, so exclude them temporarily.
see #40873, #40874

Reply:
This should probably be done with a better filter. Since this test needs null pointer exception detection, we should filter that out here instead of relying on platform excludes.